rir/src/interpreter/interp.cpp

```diff
@@ -1509,6 +1512,21 @@ void deoptFramesWithContext(InterpreterInstance* ctx,
 SEXP evalRirCode(Code* c, InterpreterInstance* ctx, SEXP env,
                  const CallContext* callCtxt, Opcode* initialPC,
                  R_bcstack_t* localsBase) {
+    if (c->nativeCode) {
+        if (!callCtxt || callCtxt->hasStackArgs() ||
+            (callCtxt->suppliedArgs == 0 &&
+             callCtxt->givenAssumptions.includes(
+                 Assumption::NotTooFewArguments))) {
+            void* args = callCtxt ? (void*)callCtxt->stackArgs : (void*)0xdead;
+            return ((SEXP(*)(Code*, InterpreterInstance*, void*, SEXP,
```

Comment:
Whoa, this is ugly. Can you split this into two lines or something, i.e. one to cast `c->nativeCode` and one to call its result?

Reply:
hehe, thanks, yeah, this is all still pretty much written in anger... I'll do another pass, and let ppl know if it's ready for an actual review. I am just expecting to also hit a couple of roadblocks making this work on CI that's why I started with this PR.